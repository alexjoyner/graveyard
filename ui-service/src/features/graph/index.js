import React from 'react';
import 'd3';
import c3 from 'c3';
import 'c3/c3.css';
import {ErrorBoundary} from "../error-boundary/index";

export class Graph extends React.Component {
    constructor(props) {
        super(props);
        this.chart = {};
    }
    componentDidMount() {
        this.updateChart();
    }
    shouldComponentUpdate(nextProps) {
        this.chart.load({
            columns: nextProps.chartOpts.data.columns,
        });
        return false;
    }
    componentDidUpdate() {
        this.updateChart();
    }

    updateChart() {
        this.chart = c3.generate({
            bindto: `#${this.props.chartID}`,
            ...this.props.chartOpts,
        });
    }

    render() {
        return (
            <ErrorBoundary>
                <div id={this.props.chartID} ></div>
            </ErrorBoundary>
        );
    }
}