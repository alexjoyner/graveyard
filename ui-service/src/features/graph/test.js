import React from 'react';
import {shallow} from 'enzyme';
import {Graph} from ".";


describe('Features/Graph', () => {
    let component;
    beforeEach(() => {
        component = shallow(<Graph />);
    });
    it('renders without exploding', () => {
        expect(component.length).toEqual(1);
    });
    describe('analog gauge graph', () => {
        it('should render if called with analog gauge as prop type');
    });
    describe('line graph', () => {
        it('should render if called with line graph as prop type');
    });
});