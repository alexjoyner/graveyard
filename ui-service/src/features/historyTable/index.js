import React from 'react';
import 'purecss';

export class HistoryTable extends React.Component {

    constructor(props) {
        super(props);
        this.getNoDataRow = () => {
            return;
        }
    }

    render() {
        return (
            <div>
                <table className={"pure-table"}>
                    <thead>
                    <tr>
                        <th>Timestamp</th>
                        <th>Value</th>
                    </tr>
                    </thead>
                    <tbody>
                    <tr>
                        <td colSpan={2}>Sorry, Nothing to display</td>
                    </tr>
                    </tbody>
                </table>
            </div>
        );
    }
}