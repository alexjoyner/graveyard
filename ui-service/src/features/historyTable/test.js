import React from 'react';
import { shallow } from 'enzyme';
import {HistoryTable} from "./index";


describe('Feature/History Table', () => {
    let component;
    beforeEach(() => {
        component = shallow(<HistoryTable />);
    });
   it('renders without exploding', () => {
       expect(component.length).toEqual(1);
   });
   describe('History table styles', () => {
       it('contains a .pure-table', () => {
           expect(component.find('.pure-table').length).toEqual(1);
       });
   });
   it('contains a single table', () => {
      expect(component.find('table').length).toEqual(1);
   });
   describe('columns', () => {
       it('has only two columns', () => {
           expect(component.find('th').length).toEqual(2);
       });
       it('has a column named timestamp', () => {
           expect(component.contains(<th>Timestamp</th>)).toEqual(true);
       });
       it('has a column named value', () => {
           expect(component.contains(<th>Value</th>)).toEqual(true);
       });
   });
   describe('rows', () => {
       it('has a function to get default row when no data is passed', () => {
       });
       it('has a default row when their is no data to display', () => {
           expect(component.contains(
               <tbody>
               <tr>
                   <td colSpan={2}>Sorry, Nothing to display</td>
               </tr>
               </tbody>
           )).toEqual(true);
       });
       it('displays the correct values from a state with data');
   })
});