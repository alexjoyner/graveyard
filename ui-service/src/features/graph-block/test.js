import React from 'react';
import { shallow } from 'enzyme';
import { GraphBlock} from "./index";
import { Graph } from '../graph';
import { HistoryTable } from '../historyTable';
describe('Feature/GraphBlock', () => {
    let component;
    beforeEach(() => {
        component = shallow(<GraphBlock />);
    });
    describe('GraphBlock styles', () => {
       it('should have a class of .GraphBody', () => {
          expect(component.find('.GraphBody').length).toEqual(1);
       });
    });
    it('renders without exploding', () => {
        expect(component.length).toEqual(1);
    });
    it('displays a history table', () => {
        expect(component.find(HistoryTable).length).toEqual(0);
    });
    it('displays two graphs', () => {
        expect(component.find(Graph).length).toEqual(2);
    });
    it('displays an analog gauge graph');
    it('displays a bar graph');
});
