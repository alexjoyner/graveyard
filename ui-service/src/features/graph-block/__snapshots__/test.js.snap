// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature/GraphBlock renders without exploding 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GraphBlock
    BlockID="block0"
    data={
        Object {
            "type": "test sensor",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          test sensor
</h5>,
        <div
          className="pure-g"
>
          <div
                    className="pure-u-1-1"
          >
                    <Graph
                              chartID="gauge-block0"
                              chartOpts={
                                        Object {
                                                  "color": Object {
                                                    "pattern": Array [
                                                      "#50bcef",
                                                      "#F6C600",
                                                      "#4eac5b",
                                                      "#FF0000",
                                                    ],
                                                    "threshold": Object {
                                                      "values": Array [
                                                        30,
                                                        70,
                                                        78,
                                                        100,
                                                      ],
                                                    },
                                                  },
                                                  "gauge": Object {
                                                    "label": Object {
                                                      "format": [Function],
                                                    },
                                                  },
                                                }
                              }
                    />
          </div>
</div>,
      ],
      "className": "GraphBody",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test sensor",
        },
        "ref": null,
        "rendered": "test sensor",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="pure-u-1-1"
>
            <Graph
                        chartID="gauge-block0"
                        chartOpts={
                                    Object {
                                                "color": Object {
                                                  "pattern": Array [
                                                    "#50bcef",
                                                    "#F6C600",
                                                    "#4eac5b",
                                                    "#FF0000",
                                                  ],
                                                  "threshold": Object {
                                                    "values": Array [
                                                      30,
                                                      70,
                                                      78,
                                                      100,
                                                    ],
                                                  },
                                                },
                                                "gauge": Object {
                                                  "label": Object {
                                                    "format": [Function],
                                                  },
                                                },
                                              }
                        }
            />
</div>,
          "className": "pure-g",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Graph
              chartID="gauge-block0"
              chartOpts={
                            Object {
                                          "color": Object {
                                            "pattern": Array [
                                              "#50bcef",
                                              "#F6C600",
                                              "#4eac5b",
                                              "#FF0000",
                                            ],
                                            "threshold": Object {
                                              "values": Array [
                                                30,
                                                70,
                                                78,
                                                100,
                                              ],
                                            },
                                          },
                                          "gauge": Object {
                                            "label": Object {
                                              "format": [Function],
                                            },
                                          },
                                        }
              }
/>,
            "className": "pure-u-1-1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "chartID": "gauge-block0",
              "chartOpts": Object {
                "color": Object {
                  "pattern": Array [
                    "#50bcef",
                    "#F6C600",
                    "#4eac5b",
                    "#FF0000",
                  ],
                  "threshold": Object {
                    "values": Array [
                      30,
                      70,
                      78,
                      100,
                    ],
                  },
                },
                "gauge": Object {
                  "label": Object {
                    "format": [Function],
                  },
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            test sensor
</h5>,
          <div
            className="pure-g"
>
            <div
                        className="pure-u-1-1"
            >
                        <Graph
                                    chartID="gauge-block0"
                                    chartOpts={
                                                Object {
                                                            "color": Object {
                                                              "pattern": Array [
                                                                "#50bcef",
                                                                "#F6C600",
                                                                "#4eac5b",
                                                                "#FF0000",
                                                              ],
                                                              "threshold": Object {
                                                                "values": Array [
                                                                  30,
                                                                  70,
                                                                  78,
                                                                  100,
                                                                ],
                                                              },
                                                            },
                                                            "gauge": Object {
                                                              "label": Object {
                                                                "format": [Function],
                                                              },
                                                            },
                                                          }
                                    }
                        />
            </div>
</div>,
        ],
        "className": "GraphBody",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test sensor",
          },
          "ref": null,
          "rendered": "test sensor",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="pure-u-1-1"
>
              <Graph
                            chartID="gauge-block0"
                            chartOpts={
                                          Object {
                                                        "color": Object {
                                                          "pattern": Array [
                                                            "#50bcef",
                                                            "#F6C600",
                                                            "#4eac5b",
                                                            "#FF0000",
                                                          ],
                                                          "threshold": Object {
                                                            "values": Array [
                                                              30,
                                                              70,
                                                              78,
                                                              100,
                                                            ],
                                                          },
                                                        },
                                                        "gauge": Object {
                                                          "label": Object {
                                                            "format": [Function],
                                                          },
                                                        },
                                                      }
                            }
              />
</div>,
            "className": "pure-g",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Graph
                chartID="gauge-block0"
                chartOpts={
                                Object {
                                                "color": Object {
                                                  "pattern": Array [
                                                    "#50bcef",
                                                    "#F6C600",
                                                    "#4eac5b",
                                                    "#FF0000",
                                                  ],
                                                  "threshold": Object {
                                                    "values": Array [
                                                      30,
                                                      70,
                                                      78,
                                                      100,
                                                    ],
                                                  },
                                                },
                                                "gauge": Object {
                                                  "label": Object {
                                                    "format": [Function],
                                                  },
                                                },
                                              }
                }
/>,
              "className": "pure-u-1-1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "chartID": "gauge-block0",
                "chartOpts": Object {
                  "color": Object {
                    "pattern": Array [
                      "#50bcef",
                      "#F6C600",
                      "#4eac5b",
                      "#FF0000",
                    ],
                    "threshold": Object {
                      "values": Array [
                        30,
                        70,
                        78,
                        100,
                      ],
                    },
                  },
                  "gauge": Object {
                    "label": Object {
                      "format": [Function],
                    },
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
